packages/hub/routes/profiles.ts

```diff
@@ -0,0 +1,253 @@
+
+    if (profile) {
+      if (ctx.state.userAddress !== profile.ownerAddress) {
+        ctx.status = 403;
+        return;
+      }
+    }
+
+    if (!profile) {
+      ctx.status = 404;
+      return;
+    }
+
+    let attributes = ctx.request.body.data.attributes;
+
+    if (attributes['name']) {
+      profile.name = this.sanitizeText(ctx.request.body.data.attributes['name']);
+    }
+
+    if (attributes['color']) {
+      profile.color = this.sanitizeText(ctx.request.body.data.attributes['color']);
+    }
+
+    if (attributes['text-color']) {
+      profile.textColor = this.sanitizeText(ctx.request.body.data.attributes['text-color']);
+    }
+
+    if (attributes['profile-description']) {
+      profile.profileDescription = this.sanitizeText(ctx.request.body.data.attributes['profile-description']);
+    }
+
+    if (attributes['profile-image-url']) {
+      profile.profileImageUrl = this.sanitizeText(ctx.request.body.data.attributes['profile-image-url']);
+    }
+
+    if (attributes['links']) {
+      profile.links = ctx.request.body.data.attributes['links'];
+    }
+
+    let errors = await this.profileValidator.validate(profile);
+    let hasErrors = Object.values(errors).flatMap((i) => i).length > 0;
+
+    if (hasErrors) {
+      ctx.status = 422;
+      ctx.body = {
+        errors: serializeErrors(errors),
+      };
+    } else {
+      await prisma.profile.update({
+        data: {
+          profileDescription: profile.profileDescription,
+          profileImageUrl: profile.profileImageUrl,
+          links: profile.links,
+        },
+        where: { id: profile.id },
+      });
+
+      let serialized = await this.profileSerializer.serialize(profile);
+
+      ctx.status = 200;
+      ctx.body = serialized;
+    }
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  async getValidation(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let slug: string = ctx.params.slug;
+    let validationResult = await this.validateSlug(slug);
+
+    ctx.status = 200;
+    ctx.body = validationResult;
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  async validateSlug(slug: string) {
+    let errorMessage = validateMerchantId(slug);
+
+    if (errorMessage) {
+      return {
+        slugAvailable: false,
+        detail: errorMessage,
+      };
+    } else {
+      if (
+        this.reservedWords.isReserved(slug, (reservedWord) => reservedWord.replace(/[^0-9a-zA-Z]/g, '').toLowerCase())
+      ) {
+        return {
+          slugAvailable: false,
+          detail: 'This ID is not allowed',
+        };
+      } else {
+        let prisma = await this.prismaManager.getClient();
+        let profile = await prisma.profile.findFirst({ where: { slug } });
+        return {
+          slugAvailable: profile ? false : true,
+          detail: profile ? 'This ID is already taken. Please choose another one' : 'ID is available',
+        };
```

Comment:
Nit-ish (not suggestion because formatting)

```js
if (profile) {
  return { slugAvailable: false, detail: 'This ID is already taken. Please choose another one' }
} else {
  return { slugAvailable: true, detail: 'ID is available' }
}
```

Reply:
updated, thanks